src/wasm/wasm-binary.cpp

```diff
@@ -109,20 +117,21 @@ template<typename T> int32_t WasmBinaryWriter::startSection(T code) {
   if (sourceMap) {
     sourceMapLocationsSizeAtSectionStart = sourceMapLocations.size();
   }
+  binaryLocationsSizeAtSectionStart = binaryLocations.size();
   return writeU32LEBPlaceholder(); // section size to be filled in later
 }
 
 void WasmBinaryWriter::finishSection(int32_t start) {
   // section size does not include the reserved bytes of the size field itself
   int32_t size = o.size() - start - MaxLEB32Bytes;
   auto sizeFieldSize = o.writeAt(start, U32LEB(size));
-  if (sizeFieldSize != MaxLEB32Bytes) {
+  auto adjustment = MaxLEB32Bytes - sizeFieldSize;
```

Comment:
now that `adjustment` isn't quite as local, maybe it should be called something more descriptive such as `sectionHeaderSizeAdjustment` (or something better/shorter if you can think of something)

Reply:
Good idea, renaming to `adjustmentForLEBShrinking`, how is that?

Reply:
it's just LEB shrinking of the header though, right?

Reply:
Yes, just the header's LEB (here, of the section - the same happens in each function as they also have an LEB size).